packages/health-and-info/src/main/HealthCheckMiddleware.test.ts

```diff
@@ -0,0 +1,137 @@
+import { NextFunction, Request, Response } from 'express'
+import HealthCheckMiddleware from './HealthCheckMiddleware'
+import HealthCheck from './HealthCheck'
+import DeploymentInfo from './DeploymentInfo'
+
+import { HealthCheckOptions } from './types/HealthCheckOptions'
+
+jest.mock('./HealthCheck')
+jest.mock('./DeploymentInfo')
+
+const MockedHealthCheck = HealthCheck as jest.MockedClass<typeof HealthCheck>
+const MockedDeploymentInfo = DeploymentInfo as jest.MockedClass<typeof DeploymentInfo>
+
+describe('HealthCheckMiddleware', () => {
+  const mockHealthyService = {
+    status: 'UP',
+    components: [
+      {
+        name: 'mock-component-1',
+        status: 'UP',
+      },
+      {
+        name: 'mock-component-2',
+        status: 'UP',
+      },
+    ],
+  }
+
+  const mockUnHealthyService = {
+    status: 'DOWN',
+    components: [
+      {
+        name: 'mock-component-1',
+        status: 'UP',
+      },
+      {
+        name: 'mock-component-2',
+        status: 'DOWN',
+      },
+    ],
+  }
+
+  const mockShortDeploymentInfo = {
+    version: '1',
+    build: {
+      buildNumber: '1',
+      gitRef: '1',
+    },
+    uptime: 5,
+  }
+
+  const mockFullDeploymentInfo = {
+    git: {
+      branch: 'main',
+    },
+    build: {
+      artifact: 'test-service-name',
+      version: '1',
+      name: 'test-service-name',
+    },
+    productId: 'AL5X',
+    uptime: 5,
+  }
+
+  let mockReq: Request
+  let mockRes: Response
+  let mockNext: NextFunction
+
+  beforeEach(() => {
+    mockReq = {} as Request
+    mockRes = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn(),
+    } as unknown as Response
+    mockNext = jest.fn()
+  })
+
+  describe('health', () => {
+    it('should return 200 status with JSON if all components healthy', async () => {
+      const healthCheckMiddleware = new HealthCheckMiddleware({} as HealthCheckOptions)
+      ;(MockedHealthCheck.prototype.check as jest.Mock).mockResolvedValue(mockHealthyService)
```

Comment:
**suggestion:** rather than mocking the details of an internal dependencies in these tests, probably be a bit nicer to widen its scope from being a pure unit test and pass in health checks stubs via the constructor.

```ts
new HealthCheckMiddleware({
   applicationInfo,
   components: [
      () => mockHealthyService
   ]
})
```

The other alternative would be doing a bit IoC and making it possible to inject a HealthCheck mock but that would make the public constructor a bit murky.. in traditional library API design they'd probably hide the constructor behind a factory to allow library implementors to potentially alter implementations in the future without breaking clients but that'd probably be overkill for us.

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-typescript-lib/pull/7/commits/a7344b1d081c61a371351112ad6447d9bd7eab62, IoC option would be good refactor in future, but logic/implementation is fairly simple under the hood so this will likely suffice